arches/app/models/migrations/11613_node_nodegroup_root.py

```diff
@@ -0,0 +1,35 @@
+# Generated by Django 5.1.3 on 2024-11-11 07:27
+
+import django.db.models.deletion
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("models", "10437_node_alias_not_null"),
+    ]
+
+    def set_nodegroup_root(apps, schema_editor):
+        Node = apps.get_model("models", "Node")
+        all_but_top_nodes = Node.objects.exclude(istopnode=True)
+        for node in all_but_top_nodes:
+            assert node.nodegroup_id is not None, f"Missing nodegroup for {node!r}"
+            node.nodegroup_root_id = node.nodegroup_id
+        Node.objects.bulk_update(all_but_top_nodes, ["nodegroup_root_id"])
+
+    operations = [
+        migrations.AddField(
+            model_name="node",
+            name="nodegroup_root",
+            field=models.ForeignKey(
+                blank=True,
+                null=True,
+                on_delete=django.db.models.deletion.CASCADE,
+                related_name="sibling_nodes",
+                related_query_name="sibling_node",
+                to="models.node",
+            ),
+        ),
+        migrations.RunPython(set_nodegroup_root, migrations.RunPython.noop),
```

Comment:
I probably need to do a slightly fancier update/rewind on the serialized graphs here.

Reply:
This occurred to me when I saw a failure in the UI, but grepping for where we do `Node(**serialized_node)` showed that this should be pretty-okay as is, but I did at a little safety in [0304a86](https://github.com/archesproject/arches/pull/11614/commits/0304a860373763e1282850df1297056bf443c65d) if not.

Reply:
After reversing migrations, I found the `nodegroup_root_id` property on the serialized graph caused this error: `TypeError: Node() got unexpected keyword arguments: 'nodegroup_root_id'` here:

https://github.com/archesproject/arches/blob/0304a860373763e1282850df1297056bf443c65d/arches/app/models/graph.py#L224-L225